backend/src/main/java/ai/giskard/web/rest/controllers/GalleryUnlockController.java

```
@@ -0,0 +1,36 @@
+package ai.giskard.web.rest.controllers;
+
+import ai.giskard.config.ApplicationProperties;
+import ai.giskard.service.InitService;
+import ai.giskard.web.dto.GalleryUnlockDTO;
+import lombok.RequiredArgsConstructor;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Objects;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v2/hfs/unlock")
+public class GalleryUnlockController {
+    private final ApplicationProperties applicationProperties;
+
+    @GetMapping("")
+    public GalleryUnlockDTO getUnlockStatus() {
+        GalleryUnlockDTO unlockDTO = new GalleryUnlockDTO();
+        unlockDTO.setUnlocked(InitService.isUnlocked());
+        unlockDTO.setToken(null);
+        return unlockDTO;
+    }
+
+    @PostMapping("")
+    public GalleryUnlockDTO setUnlockStatus(@RequestBody GalleryUnlockDTO unlockDTO) {
+        if (StringUtils.hasText(unlockDTO.getToken()) &&
```

We should probably throw an exception is the token is not provided/wrong